### PR TITLE
Add non Litinski Operations

### PR DESCRIPTION
**Pull request by gwwatkin**
This is part of the bigger task #96.

Initially we thought in #91 and #93 to add just Transversal operations, but in the Horsman et al. paper there is a precise protocol for CNOT that doesn't require converting to PauliRotations before lattice surgery. We could use that directly. That means adding another kind of operation, which is  neither transversal nor Litinski. It's the original kind of lattice surgery.

### CODE REVIEW

File: circuit.py [hunk 1]
```diff
@@ -4,24 +4,25 @@
...
-        Generating a circuit
+        Generating a Pauli circuit
 
         Args:
             no_of_qubit (int): Number of qubits in the circuit
             name (str, optional): Circuit's name (for display). Defaults to ''.
         """
         self.qubit_num:     int = no_of_qubit
-        self.ops:           List[PauliProductOperation] = list()
+        self.ops:           List[Union[Circuit, TransversalOp]] = list()
```
Comment (gwwatkin) [inline circuit.py:25]: In #93 we decided to go with:
Screenshot_20210512_070859
I thought that would have meant leaving the PauliCircuit class separate from transversal operations. So the mixing of transversal and non transversal would happen outside of the circuit class of which we would take segments :thinking: . Is there a particular reason why you went this way?

**Comment by alexnguyenn** [comment]
@gwwatkin Any idea on the status of this? Do we still need this feature?

**Comment by gwwatkin** [comment]
@alexnguyenn I would say it's been on hold for some time. Now I'm inclined to replace it with #116 as a more more relevant and up to date approach to direct compiling

**Comment by gwwatkin** [comment]
Updated #96 indicating #116 - compiling directly ZX to Lattice surgery as higher priority. If we want to implement Horsman style operations we'll have to rewrite this anyways because of more recent changes. There will still be a trace of this PR in #96 conversation if we need inspiration, so I think we can safely close.

**Pull request closed without merging**